src/dataworker/Dataworker.ts

```diff
@@ -731,12 +496,260 @@ export class Dataworker {
     );
   }
 
-  async validateRootBundle(poolRebalanceRoot: string, relayerRefundRoot: string, slowRelayRoot: string) {
-    // Look at latest propose root bundle event earlier than a block number.
-    // Determine start block numbers for end block numbers emitted in ProposedRootBundle event using
-    // HubPoolClient.getNextBundleStartBlockNumber and passing in the mainnet block of the propose root bundle txn
-    // Construct roots locally using class functions and compare with the event we found earlier.
-    // If any roots mismatch, pinpoint the errors to give details to the caller.
+  async validatePendingRootBundle(submitDispute = true) {
+    if (!this.clients.hubPoolClient.isUpdated) throw new Error(`HubPoolClient not updated`);
+    const hubPoolChainId = (await this.clients.hubPoolClient.hubPool.provider.getNetwork()).chainId;
+
+    // Exit early if a bundle is not pending.
+    if (!this.clients.hubPoolClient.hasPendingProposal()) {
+      this.logger.debug({
+        at: "Dataworker#validate",
+        message: "No pending proposal, nothing to validate",
+      });
+      return;
+    }
+
+    const pendingRootBundle = this.clients.hubPoolClient.getPendingRootBundleProposal();
+    this.logger.debug({
+      at: "Dataworker#validate",
+      message: "Found pending proposal",
+      pendingRootBundle,
+    });
+
+    // Exit early if challenge period timestamp has passed:
+    if (this.clients.hubPoolClient.currentTime > pendingRootBundle.challengePeriodEndTimestamp) {
+      this.logger.debug({
+        at: "Dataworke#validater",
+        message: "Challenge period passed, cannot dispute",
+      });
+      return;
+    }
```

Comment:
These are all already collapsable lines.
```suggestion
   // Exit early if a bundle is not pending.
    if (!this.clients.hubPoolClient.hasPendingProposal()) {
      this.logger.debug({ at: "Dataworker#validate", message: "No pending proposal, nothing to validate" });
      return;
    }

    const pendingRootBundle = this.clients.hubPoolClient.getPendingRootBundleProposal();
    this.logger.debug({ at: "Dataworker#validate", message: "Found pending proposal", pendingRootBundle });

    // Exit early if challenge period timestamp has passed:
    if (this.clients.hubPoolClient.currentTime > pendingRootBundle.challengePeriodEndTimestamp) {
      this.logger.debug({ at: "Dataworke#validater", message: "Challenge period passed, cannot dispute" });
      return;
    }
```